src/pkgcheck/checks/reserved.py

```diff
@@ -0,0 +1,100 @@
+import re
+from .. import addons, bash, results, sources
+from . import Check
+
+
+reserved_prefixes = ('__', 'abort', 'dyn', 'prep')
+reserved_substrings = ('hook', 'paludis')  # 'ebuild' is special case
+reserved_ebuild_regex = re.compile(r'.*[^a-zA-Z]ebuild.*')
```

Comment:
I feel like you could create a base class that holds these properties instead of defining them as global variables. Like:

```python
class ReservedNameCheck(Check):
    reserved_prefixes = ('__', 'abort', 'dyn', 'prep')
    reserved_substrings = ('hook', 'paludis')  # 'ebuild' is special case
    reserved_ebuild_regex = re.compile(r'.*[^a-zA-Z]ebuild.*')


class EclassReservedNameCheck(ReservedNameCheck):
    ...
    def feed(self, eclass):
        # use self.reserved_prefixes, etc here
        pass


class EbuildReservedNameCheck(ReservedNameCheck):
    ...
    def feed(self, pkg):
        # use self.reserved_prefixes, etc here
        pass
```

Reply:
Thanks, in the end I joined the main logic between the checks into a middle one.

Reply:
Even better! :-)